BungeeCord-Patches/0049-Add-some-things-to-the-API.patch

```diff
@@ -0,0 +1,264 @@
++     */
++    public Title getCurrentTitle();
++
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java b/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java
+index 2e5ae4fb..981dd8ef 100644
+--- a/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java
++++ b/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java
+@@ -67,6 +67,13 @@ public class Plugin
+         return new File( getProxy().getPluginsFolder(), getDescription().getName() );
+     }
+ 
++    // Waterfall start - allow plugins to get plugin names directly
++    public String getName()
++    {
++        return description.getName();
++    }
++    // Waterfall end
++
+     /**
+      * Get a resource from within this plugins jar or container. Care must be
+      * taken to close the returned stream.
+diff --git a/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java b/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java
+index 81bd18f0..44ff7cd9 100644
+--- a/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java
++++ b/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java
+@@ -181,10 +181,14 @@ public class PluginManager
+             {
+                 if ( proxy.getConfig().isLogCommands() )
+                 {
+-                    proxy.getLogger().log( Level.INFO, "{0} executed command: /{1}", new Object[]
++                    // Waterfall - check if the command sender is not the console
++                    if ( !sender.getName().equalsIgnoreCase( proxy.getConsole().getName() ) )
+                     {
+-                        sender.getName(), commandLine
+-                    } );
++                        proxy.getLogger().log( Level.INFO, "{0} executed command: /{1}", new Object[]
++                                {
++                                        sender.getName(), commandLine
++                                } );
++                    }
+                 }
+                 command.execute( sender, args );
+             } else if ( commandLine.contains( " " ) && command instanceof TabExecutor )
+diff --git a/api/src/main/java/net/md_5/bungee/api/scheduler/TaskScheduler.java b/api/src/main/java/net/md_5/bungee/api/scheduler/TaskScheduler.java
+index 5d2b088c..4fa6ec45 100644
+--- a/api/src/main/java/net/md_5/bungee/api/scheduler/TaskScheduler.java
++++ b/api/src/main/java/net/md_5/bungee/api/scheduler/TaskScheduler.java
+@@ -58,6 +58,35 @@ public interface TaskScheduler
+      */
+     ScheduledTask schedule(Plugin owner, Runnable task, long delay, TimeUnit unit);
+ 
++    // Waterfall start - allow scheduling tasks with ticks
++
++    /**
++     * Schedules a task to be executed asynchronously after the specified delay
++     * is up.
++     *
++     * @param owner the plugin owning this task
++     * @param task the task to run
++     * @param delay the delay before this task will be executed, specified in ticks
++     * @return the scheduled task
++     */
```

Comment:
Bungee (and by extension, Waterfall) doesn't have a concept of ticks. This is only used in NMS based servers. You can pass 50 millis if you want to run a task every tick.

Reply:
In addition, you should specify either in Javadocs or the method name (or both) that this method accepts ticks. It isn't obvious what unit the delay is in unless you read the implementation code.